### DISCUSSION TITLE
Need help with Background event playback

### DISCUSSION THREAD

**Discussion by bowerhaus** [artisan-roaster-scope/artisan#1213] [Q&A] [answered]
Hi,

I have a home built roaster that uses Phidgets for the BT sensor and for the heater which is driven via an SSR. The heater is controlled by the Artisan software PID and all seems to work fine. 

However, I am trying to control the roast using a Background profile built with the Designer. I only want to play back the DE/FC/DROP events and I want to do this via Time (not Temperature). I'm pretty sure I had this working by Temperature previously but now I don't seem to be able to playback any events at all. I have checked that Playback is on by toggling with the J key and I see PLAYBACK ON on the display.

I have attached my setup (TS0.alog) file and the background profile (TP0.alog). This profile is just a dummy test profile created so the events could be tested on my roaster without performing an entire roast. A more realistic profile that also doesn't work is attached as TP5-D16.alog.

I am using Artisan 2.8.2 because I'm on an RPi and there is no later release available for this platform as yet.

I didn't want to enter this as an Issue just yet because I'm sure I'm doing something wrong rather than it being a problem with the software. Thanks in advance for any help.

Best regards

[TP0.alog.txt](https://github.com/artisan-roaster-scope/artisan/files/11900540/TP0.alog.txt)
[TP5-D16.alog.txt](https://github.com/artisan-roaster-scope/artisan/files/11900543/TP5-D16.alog.txt)
[TS0.aset.txt](https://github.com/artisan-roaster-scope/artisan/files/11900547/TS0.aset.txt)

**Comment by MAKOMO** [comment, marked as answer]
You misunderstood what Replay Events is doing. Did you read [Profile Templates](https://artisan-roasterscope.blogspot.com/2017/10/profile-templates.html)?

The replay feature is only replaying custom events like burner changes by moving the corresponding event slider once the same temperature and time is reached, NOT main event marks like DE/FC/DROP (what for!?). There is one exception: one can activate activate "Playback DROP" which will activate DROP once the same time/temp of the background DROP is reached. DE and FC can be set at the same temp as in the background ("From Background") using the AutoDRY and AutoFCs function (menu Config >> Phases).

**Reply by bowerhaus** [reply]
Yes I did read the Profile Templates help but, you're right, I completely misunderstood what the replay events does. So I must have had AutoDRY and AutoFC set in Phases at some point and, because the phases were auto aligned with the background this is why I saw those events "replayed" by temperature at some point. It's good to know what is going on because I though I was going crazy.

The reason I want to do this is this. I'm trying to build a "hot air gun style" automated roaster a bit like the IKAWA which is driven by different background roasting profiles for each bean type. I'm not that interested in DE (or DROP really) but FC would marked directly from the profile. Once FC is triggered I would then trigger a DROP automatically based on Development % (see my Q&A #1206 which you answered yesterday). So the only auto event I really need is FC.

I'm looking to trigger via Time not Temperature because, with the PID control, there is a slight oscillation of the temperature around the desired profile. I've tried tuning this out with the PID values but the example below seems about as good as I can get it. As you can see from the attached image the FC mark occurs late because the temperature is reached later but I have observed that the actual FC occurs nearly always bang on the same time. Because of the shallowness of the BT curve I can sometimes see FC being up to 40s late because of this.

Thanks very much for the reply. I guess the solution is for me to try and improve the smoothness of the PID BT curve.

Best regards

![BHR10](https://github.com/artisan-roaster-scope/artisan/assets/2306120/b4f67d43-bf07-4edf-ba3b-b93260cfd14b)
